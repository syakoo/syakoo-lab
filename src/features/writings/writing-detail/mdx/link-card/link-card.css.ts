import { style } from "@vanilla-extract/css";
import { recipe } from "@vanilla-extract/recipes";

import { theme } from "@/shared/design-system/theme.css";

export const linkCardStyles = {
  root: style({
    display: "flex",
    gap: theme.space[50],
    justifyContent: "space-between",
    marginBlock: theme.space[200],
    backgroundColor: theme.color.background.primary,
    border: `1px solid ${theme.color.background.secondary}`,
    borderRadius: theme.radius[100],
    overflow: "hidden",
  }),
  body: style({
    display: "flex",
    flexDirection: "column",
    gap: theme.space[50],
    paddingBlock: theme.space[200],
    paddingInline: theme.space[100],
  }),
  textContainer: recipe({
    base: {
      display: "-webkit-box",
      overflow: "hidden",
      WebkitLineClamp: 1,
      WebkitBoxOrient: "vertical",
    },
    variants: {
      lines: {
        2: {
          WebkitLineClamp: 2,
        },
        1: {
          WebkitLineClamp: 1,
        },
      },
    },
  }),
  thumbnailWrapper: style({
    maxWidth: "300px",
    minWidth: "100px",
  }),
  thumbnail: style({
    flexShrink: 0,
    width: "100%",
    maxHeight: "128px",
    height: "100%",
    objectFit: "cover",
    margin: 0,
  }),
};
