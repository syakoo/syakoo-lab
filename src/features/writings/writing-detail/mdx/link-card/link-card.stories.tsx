import type { Meta, StoryObj } from "@storybook/nextjs";

import { LinkCard } from ".";

const meta = {
  component: LinkCard,
  parameters: {},
} satisfies Meta<typeof LinkCard>;

export default meta;

type Story = StoryObj<typeof LinkCard>;

export const Sample: Story = {
  args: {
    imgSrc: "https://placehold.jp/3063D4/ffffff/400x300.png",
    title: "Sample Title",
    url: "https://syakoo-lab.com/",
    description: "sample description",
    faviconSrc: "https://placehold.jp/3063D4/ffffff/64x64.png",
  },
};

export const LongText: Story = {
  args: {
    imgSrc: "https://placehold.jp/3063D4/ffffff/400x300.png",
    title: "Sample Title Sample Title Sample Title Sample Title Sample Title",
    url: "https://syakoo-lab.com/",
    description:
      "sample description sample description sample description sample description",
    faviconSrc: "https://placehold.jp/3063D4/ffffff/64x64.png",
  },
};
