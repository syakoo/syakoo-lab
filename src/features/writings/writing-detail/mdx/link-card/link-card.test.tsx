import { testStories } from "@/test/test-stories";

import * as stories from "./link-card.stories";

describe("LinkCard", () => {
  testStories(stories);
});
