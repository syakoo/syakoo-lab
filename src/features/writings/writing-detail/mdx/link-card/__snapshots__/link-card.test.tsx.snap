// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCard スナップショット LongText 1`] = `
<body>
  <div>
    <a
      class="link_linkStyle__1vz8ftz0 link_linkStyle_display_block__1vz8ftz4"
      href="https://syakoo-lab.com/"
    >
      <div
        class="link-card_linkCardStyles_root__1is99uj0"
      >
        <div
          class="link-card_linkCardStyles_body__1is99uj1"
        >
          <div
            class="link-card_linkCardStyles_textContainer__1is99uj2 link-card_linkCardStyles_textContainer_lines_2__1is99uj4"
          >
            <span
              class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_100__1djq151l text_textStyle_variants_size_100__1djq1516 text_textStyle_weight_bold__1djq151t"
            >
              Sample Title Sample Title Sample Title Sample Title Sample Title
            </span>
          </div>
          <div
            class="link-card_linkCardStyles_textContainer__1is99uj2 link-card_linkCardStyles_textContainer_lines_1__1is99uj3"
          >
            <span
              class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
            >
              sample description sample description sample description sample description
            </span>
          </div>
          <div
            class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
          >
            <img
              alt=""
              height="14"
              src="https://placehold.jp/3063D4/ffffff/64x64.png"
              width="14"
            />
            <span
              class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
            >
              syakoo-lab.com
            </span>
          </div>
        </div>
        <div
          class="link-card_linkCardStyles_thumbnailWrapper__1is99uj5"
        >
          <img
            alt="Sample Title Sample Title Sample Title Sample Title Sample Title"
            class="link-card_linkCardStyles_thumbnail__1is99uj6"
            src="https://placehold.jp/3063D4/ffffff/400x300.png"
          />
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`LinkCard スナップショット Sample 1`] = `
<body>
  <div>
    <a
      class="link_linkStyle__1vz8ftz0 link_linkStyle_display_block__1vz8ftz4"
      href="https://syakoo-lab.com/"
    >
      <div
        class="link-card_linkCardStyles_root__1is99uj0"
      >
        <div
          class="link-card_linkCardStyles_body__1is99uj1"
        >
          <div
            class="link-card_linkCardStyles_textContainer__1is99uj2 link-card_linkCardStyles_textContainer_lines_2__1is99uj4"
          >
            <span
              class="text_textStyle__1djq151e text_textStyle_color_currentColor__1djq151i text_textStyle_variants_color_currentColor__1djq1513 text_textStyle_size_100__1djq151l text_textStyle_variants_size_100__1djq1516 text_textStyle_weight_bold__1djq151t"
            >
              Sample Title
            </span>
          </div>
          <div
            class="link-card_linkCardStyles_textContainer__1is99uj2 link-card_linkCardStyles_textContainer_lines_1__1is99uj3"
          >
            <span
              class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
            >
              sample description
            </span>
          </div>
          <div
            class="flex_flexStyle__zqca9ti flex_flexStyle_align_center__zqca9ts flex_flexStyle_variants_align_center__zqca9t7 flex_flexStyle_direction_row__zqca9tj flex_flexStyle_gap_50__zqca9tx flex_flexStyle_variants_gap_50__zqca9tc flex_flexStyle_justify_flexStart__zqca9tl flex_flexStyle_variants_justify_flexStart__zqca9t0"
          >
            <img
              alt=""
              height="14"
              src="https://placehold.jp/3063D4/ffffff/64x64.png"
              width="14"
            />
            <span
              class="text_textStyle__1djq151e text_textStyle_color_secondary__1djq151g text_textStyle_variants_color_secondary__1djq1511 text_textStyle_size_50__1djq151j text_textStyle_variants_size_50__1djq1514 text_textStyle_weight_normal__1djq151u"
            >
              syakoo-lab.com
            </span>
          </div>
        </div>
        <div
          class="link-card_linkCardStyles_thumbnailWrapper__1is99uj5"
        >
          <img
            alt="Sample Title"
            class="link-card_linkCardStyles_thumbnail__1is99uj6"
            src="https://placehold.jp/3063D4/ffffff/400x300.png"
          />
        </div>
      </div>
    </a>
  </div>
</body>
`;
